tools/ashlar/ashlar.xml

```diff
@@ -0,0 +1,184 @@
+<tool id="ashlar" name="ASHLAR" version="@VERSION@.1" profile="17.09">
+    <description>Alignment by Simultaneous Harmonization of Layer/Adjacency Registration</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+ 
+    <expand macro="requirements"/>
+    @VERSION_CMD@
+
+    <command detect_errors="exit_code"><![CDATA[
+
+        #def clean(file,type)
+          #set name_clean = str($file.element_identifier).replace('.ome.tiff','').replace('.tiff','').replace('.tiff.','')
+
+          #if $type == "raw"
+            #set file_clean = $name_clean + ".ome.tiff"
+          #elif $type == "ffp"
+            #set file_clean = $name_clean + "_ffp.ome.tiff"
+          #elif $type == "dfp"
+            #set file_clean = $name_clean + "_dfp.ome.tiff"
+          #end if
+
+          #return $file_clean
+        #end def
+
+        ## Link the illumination files to appropriate file extension
+        #for $dfp in $ldfp:
+            ln -s "$dfp" "$clean($dfp,"dfp")" &&
+        #end for
+        #for $ffp in $lffp:
+            ln -s "$ffp" "$clean($ffp,"ffp")" &&
+        #end for
+
+        @CMD_BEGIN@
+
+        ## Supply the raw images
+        #for $raw in $lraw:
+            "$raw"
+        #end for
+
+        ## Additional arguments
+        -m $max_shift
+
+        #if $flip_x
+        --flip-x
+        #end if
+
+        #if $flip_y
+        --flip-y
+        #end if
+
+        -c $adv.align_channel
+
+        #if $adv.filter_sigma
+        --filter-sigma $adv.filter_sigma
+        #end if
+
+        #if $adv.tile_size
+        --tile-size $adv.tile_size
+        #end if
+
+        --ffp
+        #for $ffp in $lffp:
+            "$clean($ffp,"ffp")"
+        #end for
+
+        --dfp
+        #for $dfp in $ldfp:
+            "$clean($dfp,"dfp")"
+        #end for
+
+        #if $adv.pyramid
+        --pyramid
+        #end if
+
+        #if $adv.flip_mosaic_x
+        --flip-mosaic-x
+        #end if
+
+        #if $adv.flip_mosaic_y
+        --flip-mosaic-y
+        #end if
+
+        -f registered.ome.tif;
+
+        #if $upgrade.decide == "do_upgrade"
+        python3 ${__tool_directory__}/pyramid_upgrade.py
+        registered.ome.tif
+
+        #if $upgrade.markers_file
+        -n `echo \$(cat $upgrade.markers_file | tail -n +2 | awk -F, '{print \$3}')`;
+        #end if
+        #end if
+    ]]></command>
+
+    <inputs>
+        <param name="lraw" type="data_collection" format="ome.tiff,tiff" collection_type="list" label="Raw Images"/>
```

Comment:
This tool can also work on a single file, isn't it? Why not using "type="data" multiple="true""?

Reply:
multiple=true mainly allows you to put in collections AND single datasets. If you want to disallow single datasets that is fine, just unusual.